stdlib/auth/src/main/java/org/ballerinalang/auth/ldap/nativeimpl/Authenticate.java

```diff
@@ -118,4 +120,58 @@ private boolean bindAsUser(String dn, byte[] credentials) throws NamingException
         }
         return isAuthenticated;
     }
+
+    public static boolean doAuthenticate(Strand strand, ObjectValue authStore, String userName, String password) {
+        byte[] credential = password.getBytes(Charset.forName(LdapConstants.UTF_8_CHARSET));
+        LdapConnectionContext connectionSource = (LdapConnectionContext) authStore.getNativeData(
+                LdapConstants.LDAP_CONNECTION_SOURCE);
+        DirContext ldapConnectionContext = (DirContext) authStore.getNativeData(LdapConstants.LDAP_CONNECTION_CONTEXT);
+        CommonLdapConfiguration ldapConfiguration = (CommonLdapConfiguration) authStore.getNativeData(
+                LdapConstants.LDAP_CONFIGURATION);
+        LdapUtils.setServiceName((String) authStore.getNativeData(LdapConstants.ENDPOINT_INSTANCE_ID));
+
+        if (LdapUtils.isNullOrEmptyAfterTrim(userName)) {
+            //TODO verify why both return and throw are needed
```

Comment:
What needs to be verified here?

Reply:
Previous code, return values has been set to context and also has thrown an exception. Need to verify this with the respective code owner for the reason. For the moment I commented out setting return values line.